src/log_file.rs

```diff
@@ -50,6 +49,20 @@ impl LogFile {
         );
     }
 
+    fn log_file_name(basename: &str, i: i32) -> String {
+        let mut t = if i == 0 {
+            format!("{}.log", basename)
+        } else {
+            format!("{}_{:03}.log", basename, i)
+        };
+        // Reduce filename's length below ecryptfs limits
+        if t.len() > ECRYPTFS_FILE_NAME_LENGTH_LIMIT {
+            let range_end = t.len() - ECRYPTFS_FILE_NAME_LENGTH_LIMIT;
+            t.replace_range(0..=range_end, "");
```

Comment:
Will this cut off the `i` and the `.log`?

Also I think there is a `String::truncate` you could use.

Reply:
No. It's only applied when name too long and there is no chance that it catches `i`

Reply:
Ah of course, you're truncating from the left. I see the source file name is at the end too so that's probably fine.